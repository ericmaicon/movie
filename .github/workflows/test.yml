name: Lint & Test

on: push

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
        with:
          node-version: '14'
      - name: Cache dependencies
        uses: actions/cache@v2.1.4
        id: cache-dependencies
        with:
          path: |
            node_modules
            */*/node_modules
          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
      - name: Install Dependencies
        if: steps.cache-dependencies.outputs.cache-hit != 'true'
        run: yarn install && yarn bootstrap
      - name: Link
        run: yarn lerna link
      - name: Build deps
        run: yarn build
      - name: Lint
        run: yarn lint
      - name: Run Tests
        run: yarn test
      - uses: codecov/codecov-action@v1
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          directory: packages/**/coverage
          fail_ci_if_error: true
