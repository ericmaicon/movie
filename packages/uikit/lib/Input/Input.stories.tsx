import React from 'react';
import { FaSearch } from 'react-icons/fa';

import { Story, Meta } from '@storybook/react';

import Input, { InputProps } from './Input';

export default {
  title: 'Form/Input',
  component: Input,
} as Meta;

const Template: Story<InputProps> = (args) => <Input {...args} />;

export const Initial = Template.bind({});
Initial.args = {
  placeholder: 'Search for a movie...',
};

export const Disabled = Template.bind({});
Disabled.args = {
  disabled: true,
};

export const Invalid = Template.bind({});
Invalid.args = {
  valid: false,
};

export const LeftIcon = Template.bind({});
LeftIcon.args = {
  startIcon: <FaSearch />,
};
