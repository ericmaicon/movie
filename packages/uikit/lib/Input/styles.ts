import styled, { css } from 'styled-components';

export type TextFieldContainerProps = {
  /**
   * If `true`, the input will have `focus`.
   */
  focus: boolean;
  /**
   * If `true`, the input will be disabled.
   */
  disabled: boolean;
  /**
   * Is the input valid?
   */
  valid?: boolean;
};

export const IconContainer = styled.div`
  align-items: center;
  display: flex;
  height: 100%;
  color: ${({ theme }) => theme.colors.black};
  padding-left: 10px;
`;

export const TextField = styled.input`
  background: transparent;
  border: none;
  color: ${({ theme }) => theme.colors.black};
  font-size: 0.2rem;
  height: 100%;
  padding: 10px 20px;
  margin: 0px;
  width: 100%;
  outline: none;
  ${({ theme }) => theme.typography.text};

  //tablet
  @media (min-width: 768px) and (max-width: 1024px) {
    font-size: 0.5rem;
  }

  //desktop
  @media (min-width: 1025px) {
    font-size: 1rem;
  }

  &:disabled {
    cursor: not-allowed;
  }

  ::placeholder {
    color: ${({ theme }) => theme.colors.black};
    font-style: italic;
  }
`;

const focusModifier = () => css`
  box-shadow: 0px 0px 0px 2px rgba(0, 0, 0, 0.2);
`;

const invalidContainerModifier = () => css`
  border: 2px solid ${({ theme }) => theme.colors.red};

  input {
    color: ${({ theme }) => theme.colors.red};
  }
`;

const disabledModifier = () => css`
  background: ${({ theme }) => theme.colors.gray};
`;

export const TextFieldContainer = styled.div<TextFieldContainerProps>`
  align-items: center;
  background: ${({ theme }) => theme.colors.white};
  border: 1px solid ${({ theme }) => theme.colors.black};
  box-sizing: border-box;
  border-radius: 25px;
  display: flex;
  flex-direction: horizontal;
  height: 30px;

  ${({ focus }) => focus && focusModifier};
  ${({ valid }) => !valid && invalidContainerModifier};
  ${({ disabled }) => disabled && disabledModifier};
`;
