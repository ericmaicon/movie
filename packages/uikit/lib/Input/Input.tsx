import React, {
  InputHTMLAttributes,
  ReactNode,
  FocusEvent,
  useState,
} from 'react';

import * as Styled from './styles';

export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  /**
   * Element placed on the left side.
   */
  startIcon?: ReactNode;
  /**
   * Wether the value is valid or not.
   */
  valid?: boolean;
}

const Input = ({
  disabled = false,
  valid = true,
  startIcon: startIconProp,
  onFocus,
  onBlur,
  ...rest
}: InputProps) => {
  const [focus, setFocus] = useState(false);

  const startIcon = startIconProp && (
    <Styled.IconContainer>{startIconProp}</Styled.IconContainer>
  );

  function handleOnFocus(event: FocusEvent<HTMLInputElement>) {
    setFocus(true);

    /* istanbul ignore else */
    if (onFocus) {
      onFocus(event);
    }
  }

  function handleOnBlur(event: FocusEvent<HTMLInputElement>) {
    setFocus(false);

    /* istanbul ignore else */
    if (onBlur) {
      onBlur(event);
    }
  }

  return (
    <>
      <Styled.TextFieldContainer
        focus={focus}
        valid={valid}
        disabled={disabled}
      >
        {startIcon}
        <Styled.TextField
          disabled={disabled}
          onFocus={handleOnFocus}
          onBlur={handleOnBlur}
          {...rest}
        />
      </Styled.TextFieldContainer>
    </>
  );
};

export default Input;
