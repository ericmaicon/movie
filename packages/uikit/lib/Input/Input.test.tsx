import React from 'react';
import { act } from 'react-dom/test-utils';
import { FaSearch } from 'react-icons/fa';

import snapshotDiff from 'snapshot-diff';

import { fireEvent, render } from '~/test/index';

import Input from './Input';

const handleChange = () => jest.fn();

describe('Input', () => {
  it('should render correctly', () => {
    const { getByTestId } = render(
      <Input value="input value" data-testid="input" onChange={handleChange} />,
    );

    const inputElement = getByTestId('input');

    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveValue('input value');
    expect(inputElement).toBeEnabled();
    expect(inputElement).toMatchSnapshot();
  });

  it('should be disabled', () => {
    const { rerender, asFragment, getByTestId } = render(
      <Input value="input value" data-testid="input" onChange={handleChange} />,
    );

    const enabled = asFragment();
    rerender(
      <Input
        disabled
        value="input value"
        data-testid="input"
        onChange={handleChange}
      />,
    );
    const disabled = asFragment();

    expect(getByTestId('input')).toBeDisabled();
    expect(disabled).toMatchSnapshot();
    expect(snapshotDiff(enabled, disabled)).toMatchSnapshot();
  });

  it('should be invalid', () => {
    const { rerender, asFragment } = render(
      <Input value="input value" data-testid="input" onChange={handleChange} />,
    );

    const valid = asFragment();
    rerender(
      <Input
        valid={false}
        value="input value"
        data-testid="input"
        onChange={handleChange}
      />,
    );
    const invalid = asFragment();

    expect(invalid).toMatchSnapshot();
    expect(snapshotDiff(valid, invalid)).toMatchSnapshot();
  });

  it('should render a left icon', () => {
    const { getAllByTestId } = render(
      <Input
        startIcon={<FaSearch data-testid="search-icon" />}
        value="input value"
        onChange={handleChange}
      />,
    );

    const [inputElement] = getAllByTestId('search-icon');
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toMatchSnapshot();
  });

  it('can have onFocus', () => {
    const handleFocus = jest.fn();

    const { getByTestId } = render(
      <Input
        value="input value"
        data-testid="input"
        onChange={handleChange}
        onFocus={handleFocus}
      />,
    );

    const textFieldElement = getByTestId('input');

    expect(textFieldElement).toBeInTheDocument();

    act(() => {
      fireEvent.keyDown(document.body, { key: 'TAB' });
      textFieldElement.focus();
    });

    expect(handleFocus).toHaveBeenCalledTimes(1);
    expect(textFieldElement).toHaveFocus();
    expect(textFieldElement).toMatchSnapshot();
  });

  it('can have onBlur', () => {
    const handleBlur = jest.fn();

    const { getByTestId } = render(
      <Input
        value="input value"
        data-testid="input"
        onChange={handleChange}
        onBlur={handleBlur}
      />,
    );

    const textFieldElement = getByTestId('input');

    expect(textFieldElement).toBeInTheDocument();

    act(() => {
      fireEvent.blur(textFieldElement);
      textFieldElement.blur();
    });

    expect(handleBlur).toHaveBeenCalledTimes(1);
    expect(textFieldElement).not.toHaveFocus();
    expect(textFieldElement).toMatchSnapshot();
  });
});
