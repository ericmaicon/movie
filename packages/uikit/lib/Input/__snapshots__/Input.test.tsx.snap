// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input can have onBlur 1`] = `
.c0 {
  background: transparent;
  border: none;
  color: #1C1C1C;
  font-size: 0.2rem;
  height: 100%;
  padding: 10px 20px;
  margin: 0px;
  width: 100%;
  outline: none;
  font-family: Roboto;
  font-weight: 300;
  font-size: 0.2rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0::-webkit-input-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c0::-moz-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c0:-ms-input-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c0::placeholder {
  color: #1C1C1C;
  font-style: italic;
}

@media (min-width:768px) and (max-width:1024px) {
  .c0 {
    font-size: 0.5rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    font-size: 1rem;
  }
}

<input
  class="c0"
  data-testid="input"
  value="input value"
/>
`;

exports[`Input can have onFocus 1`] = `
.c0 {
  background: transparent;
  border: none;
  color: #1C1C1C;
  font-size: 0.2rem;
  height: 100%;
  padding: 10px 20px;
  margin: 0px;
  width: 100%;
  outline: none;
  font-family: Roboto;
  font-weight: 300;
  font-size: 0.2rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0::-webkit-input-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c0::-moz-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c0:-ms-input-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c0::placeholder {
  color: #1C1C1C;
  font-style: italic;
}

@media (min-width:768px) and (max-width:1024px) {
  .c0 {
    font-size: 0.5rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    font-size: 1rem;
  }
}

<input
  class="c0"
  data-testid="input"
  value="input value"
/>
`;

exports[`Input should be disabled 1`] = `
<DocumentFragment>
  .c1 {
  background: transparent;
  border: none;
  color: #1C1C1C;
  font-size: 0.2rem;
  height: 100%;
  padding: 10px 20px;
  margin: 0px;
  width: 100%;
  outline: none;
  font-family: Roboto;
  font-weight: 300;
  font-size: 0.2rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
}

.c1:disabled {
  cursor: not-allowed;
}

.c1::-webkit-input-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c1::-moz-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c1:-ms-input-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c1::placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #E9E8E7;
  border: 1px solid #1C1C1C;
  box-sizing: border-box;
  border-radius: 25px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: horizontal;
  -ms-flex-direction: horizontal;
  flex-direction: horizontal;
  height: 30px;
  background: #A8B4BE;
}

@media (min-width:768px) and (max-width:1024px) {
  .c1 {
    font-size: 0.5rem;
  }
}

@media (min-width:1025px) {
  .c1 {
    font-size: 1rem;
  }
}

<div
    class="c0"
    disabled=""
  >
    <input
      class="c1"
      data-testid="input"
      disabled=""
      value="input value"
    />
  </div>
</DocumentFragment>
`;

exports[`Input should be disabled 2`] = `
"Snapshot Diff:
- First value
+ Second value

  <DocumentFragment>
    <div
-     class=\\"sc-dlfnbm pfMHK\\"
+     class=\\"sc-dlfnbm hRVNym\\"
+     disabled=\\"\\"
    >
      <input
        class=\\"sc-gsTCUz hqDQSC\\"
        data-testid=\\"input\\"
+       disabled=\\"\\"
        value=\\"input value\\"
      />
    </div>
  </DocumentFragment>"
`;

exports[`Input should be invalid 1`] = `
<DocumentFragment>
  .c1 {
  background: transparent;
  border: none;
  color: #1C1C1C;
  font-size: 0.2rem;
  height: 100%;
  padding: 10px 20px;
  margin: 0px;
  width: 100%;
  outline: none;
  font-family: Roboto;
  font-weight: 300;
  font-size: 0.2rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
}

.c1:disabled {
  cursor: not-allowed;
}

.c1::-webkit-input-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c1::-moz-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c1:-ms-input-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c1::placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #E9E8E7;
  border: 1px solid #1C1C1C;
  box-sizing: border-box;
  border-radius: 25px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: horizontal;
  -ms-flex-direction: horizontal;
  flex-direction: horizontal;
  height: 30px;
  border: 2px solid #A62218;
}

.c0 input {
  color: #A62218;
}

@media (min-width:768px) and (max-width:1024px) {
  .c1 {
    font-size: 0.5rem;
  }
}

@media (min-width:1025px) {
  .c1 {
    font-size: 1rem;
  }
}

<div
    class="c0"
  >
    <input
      class="c1"
      data-testid="input"
      value="input value"
    />
  </div>
</DocumentFragment>
`;

exports[`Input should be invalid 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,8 @@
  <DocumentFragment>
    <div
-     class=\\"sc-dlfnbm pfMHK\\"
+     class=\\"sc-dlfnbm exDEKU\\"
    >
      <input
        class=\\"sc-gsTCUz hqDQSC\\"
        data-testid=\\"input\\"
        value=\\"input value\\""
`;

exports[`Input should render a left icon 1`] = `
@media (min-width:768px) and (max-width:1024px) {

}

@media (min-width:1025px) {

}

<svg
  data-testid="search-icon"
  fill="currentColor"
  height="1em"
  stroke="currentColor"
  stroke-width="0"
  viewBox="0 0 512 512"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
  />
</svg>
`;

exports[`Input should render correctly 1`] = `
.c0 {
  background: transparent;
  border: none;
  color: #1C1C1C;
  font-size: 0.2rem;
  height: 100%;
  padding: 10px 20px;
  margin: 0px;
  width: 100%;
  outline: none;
  font-family: Roboto;
  font-weight: 300;
  font-size: 0.2rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0::-webkit-input-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c0::-moz-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c0:-ms-input-placeholder {
  color: #1C1C1C;
  font-style: italic;
}

.c0::placeholder {
  color: #1C1C1C;
  font-style: italic;
}

@media (min-width:768px) and (max-width:1024px) {
  .c0 {
    font-size: 0.5rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    font-size: 1rem;
  }
}

<input
  class="c0"
  data-testid="input"
  value="input value"
/>
`;
